### ISSUE TITLE
Can't derive an account from a new keypair after deriving an account from default keypair

### ISSUE THREAD

**Issue by qoqobolo** [status-im/status-react#20500]
# Bug Report

If you derive an account from your default keypair, then create a new keypair and try to derive an additional account from it - you will stay on the account creation page after confirmation.
It also reproduces in reverse order: first create a new keypair and derive an account from it, and then try to derive an account from the default keypair.

### Reproduction

1. Derive an account from the default keypair (`Acc1`)
2. Create a new keypair (New KP) with an account `NKP1`
3. Try to derive another account from New KP - `NKP2`

or

1. Create a new keypair (New KP) with an account `NKP1`
2. Derive another account from New KP - `NKP2`
3. Try to derive another account from your default keypair - `Acc1`

#### Expected behavior
Acc1 is created

#### Actual behavior
Nothing happens after confirmation, you stay on the ACC creation page, Acc1 is not created

Video for the second way to reproduce: 

https://github.com/status-im/status-mobile/assets/67952253/986d44bb-bfb4-4230-ad24-1cc7889fbb48

### Logs
[Status-debug-logs.zip](https://github.com/user-attachments/files/15889414/Status-debug-logs.zip)

### Additional Information

- Status version: nightly 18/07
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

**Comment by pavloburykh** [comment]
In my case, I am reproducing the the bug after creating several (3-4) derrived accounts for each key pair. So, if you will not reproduce the bug following steps from the description just keep creating new accounts within different keypairs and at some point you will stuck during creating.

**Comment by ulisesmac** [comment]
Judging by the logs, seems the problem is the derivation path suggested, I'll check it :+1: 

Update 1:

OK, according to the video, the bug is not in the derivation path suggested, since it looks consistent

**Comment by qoqobolo** [comment]
@ulisesmac update: I found a workaround for this bug in case it can be helpful for debugging.

If you faced this issue and you got stuck on the ACC creation page, try this:
1. Go to the wallet home
2. `+` > `Add account` 
3. Press `Edit` and select the key pair you're going to derive next account from
4. Press `Confirm account origin`
5. Close the acc creation page 
6. `+` > `Add account` 
Here you'll see that the selected key pair was saved and now displayed right after you open this page.
From this moment you'll be able to add this account.

If you want to derive a new acc from the default key pair, you'll need to repeat it so that the default key pair would be pre-selected when you open the acc creation page.  

https://github.com/status-im/status-mobile/assets/67952253/672875fb-bbbe-48a3-b5bd-bcb35f14218a

**Comment by ulisesmac** [comment]
> @ulisesmac update: I found a workaround for this bug in case it can be helpful for debugging.
> 
> If you faced this issue and you got stuck on the ACC creation page, try this:
> 1. Go to the wallet home
> 2. `+` > `Add account` 
> 3. Press `Edit` and select the key pair you're going to derive next account from
> 4. Press `Confirm account origin`
> 5. Close the acc creation page 
> 6. `+` > `Add account` 
> Here you'll see that the selected key pair was saved and now displayed right after you open this page.
> From this moment you'll be able to add this account.
> 
> If you want to derive a new acc from the default key pair, you'll need to repeat it so that the default key pair would be pre-selected when you open the acc creation page.  
> 
> https://github.com/status-im/status-mobile/assets/67952253/672875fb-bbbe-48a3-b5bd-bcb35f14218a
> 
> 

Thank you so much @qoqobolo !

I wasn't able to reproduce this issue, but I suspected the bug comes from the implementation of the "select key pair" screen. Now this video confirms it.

I'll try to consistently replicate it and fix it!

**Comment by ulisesmac** [comment]
@qoqobolo 

This issue drove me crazy, it was so strange, I did the reimplementation of the derivation of accounts and I found no bugs there.

The problem was the slide button, explained in simple words: 

When the button is rendered, it will perform _something_ (in this case, it will create an account for the keypair 1), but if we change the task assigned (in this case, create an account for keypair 2) it will ignore it and will keep doing the previous task.

:warning:  This is dangerous, and any other screen "changing" the task assigned, will have similar issues than this one. :warning: 

I'll try to explain what was happening behind the following video:

[01-failure.webm](https://github.com/status-im/status-mobile/assets/90291778/d0a25735-2a45-4f50-a094-0ce19ab5b2e9)

1. We have a root account with a keypair (let's call it `KP0`) :white_check_mark: 
2. We create a new KP (let's call it `KP1`) and an account for it :white_check_mark: 
3. We try create an account for `KP1` (called **NOT working account** in the video), but the button ignores we changed from `KP0` -> `KP1`, so it generates an account for `KP0` with deriv. path `m.../1` and we assign this new account to  `KP1`. I know very inconsistent!! ❌
4. We try to create a new account for `KP0` with derivation path `m.../1`, it is generated successfully, BUT it already existed inside `KP1` because of the previous step, so the app doesn't proceed and we log an error saying `"account already exists"`. ❌

Here is the video with the bug fixed:

[00-success-attempt.webm](https://github.com/status-im/status-mobile/assets/90291778/8b387e46-18cf-48aa-b27e-1b3c1cef82fb)

I'll open a PR soon :+1:
